demoapp/www/js/crashes.js

```diff
@@ -151,15 +148,13 @@ $(document).bind('pageinit', function () {
         var srcType = Camera.PictureSourceType.SAVEDPHOTOALBUM;
         var options = setOptions(srcType);
         navigator.camera.getPicture(function cameraSuccess(imageUri) {
-            if (!imageUri.length) {
-                return;
-            }
-            var index = imageUri.lastIndexOf("?");
-            attachment = imageUri.substr(0, index);
-            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, attachment);
-            updateAttachment();
-        }, function cameraError(error) {
-            alert("Unable to obtain picture: " + error);
+            binaryAttachment = imageUri;
+            attachmentsProvider.putString(attachmentsProvider.BINARY_KEY, binaryAttachment);
+            updateAttachmentUI();
+        }, function cameraError() {
+            binaryAttachment = null;
```

Comment:
I see that you're unsetting the value here but I would return the `alert` to be able to see what went wrong.

Reply:
Clicking cancel would display an error. On other platforms we cancel the dialog to unset attachment too.
You get an error at restart still.